### PR TITLE
[PIR][oneDNN] Add matmul_transpose_reshape_fuse_pass

### PR DESCRIPTION
**Pull request by LLee233**
### PR Category

Others

### PR Types

New features

### Description

Based on new pass mechanism, here we add pass "matmul_transpose_reshape_fuse_pass" for PIR.

The new pass is same as "matmul_transpose_reshape_mkldnn_fuse_pass" in "/paddle/fluid/framework/ir/mkldnn/matmul_transpose_reshapel_mkldnn_fuse_pass.cc"

### CODE REVIEW

**Review by yuanlehome — APPROVED** [review]

**Review by xinyu-intel — APPROVED** [review]

File: paddle/fluid/pir/transforms/onednn/matmul_transpose_reshape_fuse_pass.cc [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+};
+
+class MatmulTransposeReshapeFusePass : public pir::PatternRewritePass {
+ public:
+  MatmulTransposeReshapeFusePass()
+      : pir::PatternRewritePass("matmul_transpose_reshape_fuse_pass", 3) {}
+
+  pir::RewritePatternSet InitializePatterns(pir::IrContext *context) override {
+    pir::RewritePatternSet ps(context);
+    std::vector<bool> bool_set = {false, true};
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/onednn/matmul_transpose_reshape_fuse_pass.cc:252]: 没有使用吧？

Comment (LLee233) [reply paddle/fluid/pir/transforms/onednn/matmul_transpose_reshape_fuse_pass.cc:252]: 对的。。应该是之前忘删了，我在下一个PR里删掉~

File: paddle/fluid/pir/transforms/onednn/matmul_transpose_reshape_fuse_pass.cc [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    reshape({&pat.Tensor("transpose_out"), &pat.Tensor("shape")},
+            {&pat.Tensor("reshape_out"), &pat.Tensor("Xshape")});
+
+    pat.RequireNativeCall([&](const paddle::drr::MatchContext &match_ctx) {
+      std::set<bool> bool_sets = {true, false};
+      auto result_x = match_ctx.Attr<bool>("transpose_x");
+      auto result_y = match_ctx.Attr<bool>("transpose_y");
+      if (bool_sets.count(result_x) == 0 || bool_sets.count(result_y) == 0) {
+        return false;
+      }
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/onednn/matmul_transpose_reshape_fuse_pass.cc:71]: 我其实不太理解这块的逻辑，if条件不就一直为true了嘛？

Comment (LLee233) [reply paddle/fluid/pir/transforms/onednn/matmul_transpose_reshape_fuse_pass.cc:71]: 是的。。我之前以为attr不会自动check类型就加了一个check在这里。。我之后统一改掉

**Pull request merged**